datameta/static/js/submit.js

```diff
@@ -524,18 +529,162 @@ DataMeta.submit.renderMetadataHelp = function() {
     });
 }
 
+DataMeta.submit.checkBoxes = function(table, checked) {
+    document.getElementById("masterfset").disabled = true;
+    var checkboxes = document.querySelectorAll("input.datameta-entity-check[data-datameta-type='"+ table +"']")
+    checkboxes.forEach(function(box) {
+        box.checked = checked;
+        DataMeta.submit.unchecked[box.getAttribute('data-datameta-uuid')] = ! checked;
+    })
+
+    DataMeta.submit.submit(true);
+    document.getElementById("masterfset").disabled = false;
+}
+
+DataMeta.submit.deleteSelectedMeta = function() {
+    fetch(
+        "/api/ui/pending",
+        {method:"GET"}
+    ).then(function(response) {
+        if (response.ok) return response.json();
+        if (response.status==400) throw new DataMeta.AnnotatedError(request);
+    }).then(function(json) {
+        // Get checked metadatasets
+        var metadatasetIds = [];
+        json.metadatasets.forEach(function(mset) {
+            if(!DataMeta.submit.unchecked[mset.id.uuid]) { 
+                metadatasetIds.push(mset.id.uuid);
+            }
+        });
+
+        // Delete all checked metadatasets
+        fetch(DataMeta.api('rpc/delete-metadatasets'),
+        {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body:  JSON.stringify({metadatasetIds})
+        }).then(function(response) {
+            if (response.ok) {
+                return DataMeta.submit.refresh();
+            } else if (response.status==401) {
+                DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unauthenticated.", "danger")
+            } else if (response.status==403) {
+                DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Access denied.", "danger")
+            } else if (response.status==404) {
+                DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Record not found.", "danger")
+            } else if (response.status==400) {
+                response.json().then(json => {
+                    DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: " + json[0].message, "danger")
+                });
+            } else {
+                DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unknown error.", "danger")
+            }
+            DataMeta.submit.refresh();
+        }).catch(function(error) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unknown error.", "danger")
+            console.log(error)
+        });
+    }).catch(function(error) {
+        DataMeta.new_alert("<strong>ERROR</strong> An unknown error occurred when bulk deleting metadata. Please try again.", "danger")
+        console.log(error);
+    });
+}
+
+DataMeta.submit.deleteSelectedFiles = function() {
```

Comment:
Please lock the UI when you execute the actual delete action and unlock it when the action was completed. The code
```javascript
document.getElementById("masterfset").disabled = false;
```
that is still used in some places for that is deprecated, please use
```javascript
DataMeta.submit.setLock(bool);
```
instead (feel free to refactor the remaining occurrences).